R/subset_datarqst.R

```diff
@@ -37,66 +45,141 @@
+    }
 
     output_brood <- standard_data$Brood_data %>%
-      dplyr::mutate(Pop_sp = paste(PopID, Species, sep = "_")) %>%
-      dplyr::filter(Pop_sp %in% filter)
+      dplyr::mutate(Pop_sp = paste(.data$PopID, .data$Species, sep = "_")) %>%
+      dplyr::filter(.data$Pop_sp %in% filter) %>%
+      dplyr::select(-.data$Pop_sp)
 
     output_capture <- standard_data$Capture_data %>%
-      dplyr::mutate(Pop_sp = paste(CapturePopID, Species, sep = "_")) %>%
-      dplyr::filter(Pop_sp %in% filter)
+      dplyr::mutate(Pop_sp = paste(.data$CapturePopID, .data$Species, sep = "_")) %>%
+      dplyr::filter(.data$Pop_sp %in% filter) %>%
+      dplyr::select(-.data$Pop_sp)
 
     output_individual <- standard_data$Individual_data %>%
-      dplyr::mutate(Pop_sp = paste(PopID, Species, sep = "_")) %>%
-      dplyr::filter(Pop_sp %in% filter)
+      dplyr::mutate(Pop_sp = paste(.data$PopID, .data$Species, sep = "_")) %>%
+      dplyr::filter(.data$Pop_sp %in% filter) %>%
+      dplyr::select(-.data$Pop_sp)
 
     output_location <- standard_data$Location_data %>%
-      dplyr::filter(PopID %in% unique_pops)
+      dplyr::filter(.data$PopID %in% unique_pops)
 
   } else {
 
+    if(include_conflicting){
+      #Species <- c(Species, 'CCCCCC') # Use after pipelines updated to standard format 1.1
+      Species <- c(Species, 'CCCCCC', 'CONFLICTED')
+    }
+
     output_brood <- standard_data$Brood_data %>%
-      dplyr::filter(PopID %in% {{PopID}} & Species %in% {{Species}})
+      dplyr::filter(.data$PopID %in% {{PopID}} & .data$Species %in% {{Species}})
 
     output_capture <- standard_data$Capture_data %>%
-      dplyr::filter(CapturePopID %in% {{PopID}} & Species %in% {{Species}})
+      dplyr::filter(.data$CapturePopID %in% {{PopID}} & .data$Species %in% {{Species}})
 
     output_individual <- standard_data$Individual_data %>%
-      dplyr::filter(PopID %in% {{PopID}} & Species %in% {{Species}})
+      dplyr::filter(.data$PopID %in% {{PopID}} & .data$Species %in% {{Species}})
 
     if(!all(unique(Species_codes$Code) %in% Species) & all(unique(pop_names$code) %in% PopID)){
 
       PopID <- pop_species_combos %>%
-        dplyr::filter(species %in% Species) %>%
-        dplyr::pull(pop) %>%
+        dplyr::filter(.data$species %in% Species) %>%
+        dplyr::pull(.data$pop) %>%
         unique()
 
     }
 
     output_location <- standard_data$Location_data %>%
-      dplyr::filter(PopID %in% {{PopID}})
+      dplyr::filter(.data$PopID %in% {{PopID}})
 
   }
 
-  message("Output subset as .csv files...")
+  #If keeping conflicted species: remove IndvIDs that do not appear in capture data from individual data
+  #(this ensures that only data on individuals identified at least once as one of the species of interest are retained)
+  if(include_conflicting){
+    output_individual <- output_individual %>%
+      dplyr::filter(!(.data$Species %in% c('CONFLICTED', 'CCCCCC') & !(.data$IndvID %in% output_capture$IndvID)))
```

Comment:
Not sure I understand this filter. Wouldn't this work fine (i.e. remove all individuals that are now missing from capture data). It shouldn't be necessary to apply this only to conflicted individuals because all other cases will have matching individual and capture records.

```suggestion
      dplyr::filter(!.data$IndvID %in% output_capture$IndvID)
```

Reply:
Filtering now works differently, so this has been removed.